aspnetcore/blazor/components/render-modes.md

```diff
@@ -178,44 +178,48 @@ To enable global interactivity when creating a Blazor Web App:
 
 For more information, see <xref:blazor/tooling>.
 
-## Apply a render mode programatically
+## Apply a render mode in code
 
-Properties and fields can assign a render mode to a component or a group of components via inheritance.
+Properties and fields can assign a render mode.
 
-### By component definition
+The second approach described in this section, setting the render mode by component instance, is especially useful when your app specification calls for either of the following scenarios:
+
+* You have an area (folder) of the app with components that must adopt static server-side rendering (static SSR) and only run on the server. The app controls the render mode globally by setting the render mode on the `Routes` component in the `App` component based on the path to the folder.
+* You have components around the app in various locations (not in a single folder) that must adopt static SSR and only run on the server. The app controls the render mode on a per-component basis by setting the render mode with the `@rendermode` directive in component instances. Reflection is used in the `App` component to set the render mode on the `Routes` component.
```

Comment:
I'm not totally understanding why these are fundamentally different scenarios. Isn't the only difference between these what goes in the implementation for `RenderModeForPage`? i.e., in scenario 1, you decide the render mode by looking at the request path, and in scenario 2, you decide the render mode via reflection. If that's the case, could the code samples and descriptions for these be combined into one, where the potential ways to decide the render mode are explained in a single section? I could be missing something here, so let me know if I am!

Reply:
It's often best in complex use cases to format the guidance around the goal that the dev is trying to achieve. It's often critical to adopt more of a tutorial-type of approach for the new-to-programming, new-to-.NET, and new-to-Blazor folks. 

In addition to the delta on how `RenderModeForPage` is set, there are additional considerations for the force-loading layout and what the individual components need to do in regard to use of the layout and setting the render mode.

When you see me write out scenarios in this degree of detail, it's usually because I found it confusing myself and detect that readers would benefit from more of step-by-step, detailed explanation and set of examples. It's long, yes, but it's utterly clear what's going on and what the dev needs to do. If we aren't clear, it will upset readers 😡 and they'll come for the 🦖 with sharpened *TEETH* and *CLAWS* 👹😨😆. I think we'll head off doc issues by explaining it out.

Reply:
Sounds great to me, thanks for clarifying! I definitely trust your judgement about what's clear to readers, so let's go with your original strategy of having it documented as two separate scenarios 🙂

Reply:
Cool ... you still need time to fully analyze this?